src/particle/ParticleRenderer.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * This file is part of Bubba-3D.
+ *
+ * Bubba-3D is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Bubba-3D is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Bubba-3D. If not, see http://www.gnu.org/licenses/.
+ */
+
+#include "linmath/float3x3.h"
+
+#include "ResourceManager.h"
+#include "constants.h"
+
+#include "Camera.h"
+#include "ShaderProgram.h"
+#include "Particle.h"
+#include "ParticleRenderer.h"
+#include "ParticleConf.h"
+
+
+ParticleRenderer::ParticleRenderer(std::shared_ptr<Texture> texture,
+                                   std::shared_ptr<Camera> camera,
+                                   std::shared_ptr<ShaderProgram> shaderProgram)
+                                 : texture(texture),
+                                   camera(camera),
+                                   shaderProgram(shaderProgram)
+{
+    GLfloat quad[] = { //POSITION3 TEXCOORD2
+        0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+        1.0f, 1.0f, 0.0f, 1.0f, 1.0f,
+        0.0f, 1.0f, 0.0f, 0.0f, 1.0f,
+    
+        0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+        1.0f, 0.0f, 0.0f, 1.0f, 0.0f,
+        1.0f, 1.0f, 0.0f, 1.0f, 1.0f,
+    };
+    
+    glGenVertexArrays(1, &vaob);
+    glBindVertexArray(vaob);
+    
+    GLuint pos_vbo;
+    glGenBuffers(1, &pos_vbo);
+    glBindBuffer(GL_ARRAY_BUFFER, pos_vbo);
+    glBufferData(GL_ARRAY_BUFFER, sizeof(quad), quad, GL_STATIC_DRAW);
+    glEnableVertexAttribArray(0);
+    glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 5 * sizeof(GLfloat), 0);
+    
+    glEnableVertexAttribArray(1);
+    glVertexAttribPointer(1, 2, GL_FLOAT, GL_FALSE, 5 * sizeof(GLfloat), (void*)(3 * sizeof(GLfloat)));
+    
+    /* CLEANUP */
+    glBindVertexArray(0);
+    
+}
+
+ParticleRenderer::~ParticleRenderer()
+{
+
+}
+
+std::shared_ptr<ShaderProgram> ParticleRenderer::defaultShader()
+{
+    ResourceManager::loadShader("shaders/particle.vert", "shaders/particle.frag", "particleShader");
+    std::shared_ptr<ShaderProgram> shaderProgram(ResourceManager::getShader("particleShader"));
```

Comment:
before 1.0 release we should at least refactor resource manager to only return smart pointers. But if you want we can do that in another PR